lib/menu.ts

```diff
@@ -6,25 +6,30 @@
  * entirely by a custom item. For an item to be a selectable menu item, it needs `tabindex=-1`
  * attribute set. If unset, or if the "disabled" class is set, the item will not be selectable.
  *
- * Further, if `dom.data('menuItemSelected', (yesNo: boolean) => {})` is set, that callback will be
- * called whenever the item is selected and unselected. It may call onMenuItemSelected(yesNo) to keep
- * the default behavior of getting the suitable css class. A callback should be seldom needed, but it
- * is needed for nested menus.
+ * Further, if `dom.data('menuItemSelected', (yesNo: boolean, elem) => {})` is set, that callback
+ * will be called whenever the item is selected and unselected. In addition, the selected item
+ * gets a css class. A callback should be seldom needed, but it is use for nested menus.
  *
  * Clicks on items will normally propagate to the menu, where they get caught and close the menu.
  * If a click on an item should not close the menu, the item should stop the click's propagation.
  */
 import {dom, domDispose, DomElementArg, DomElementMethod, styled} from 'grainjs';
-import {Disposable, Observable, observable, onKeyDown} from 'grainjs';
+import {Disposable, onKeyDown} from 'grainjs';
 import defaultsDeep = require('lodash/defaultsDeep');
-import {IPopupContent, IPopupOptions, PopupControl, setPopupToFunc} from './popup';
+import mergeWith = require('lodash/mergeWith');
+import {IOpenController, IPopupContent, IPopupOptions, PopupControl, setPopupToFunc} from './popup';
 
-export type MenuCreateFunc = (ctl: PopupControl) => DomElementArg[];
+export type MenuCreateFunc = (ctl: IOpenController) => DomElementArg[];
 
 export interface IMenuOptions extends IPopupOptions {
-  startIndex?: number;
   isSubMenu?: boolean;
+  selectOnOpen?: boolean;
   menuCssClass?: string;    // If provided, applies the css class to the menu container.
+
+  // If given, the menu will set the `weasel-popup-open` css class on the matching ancestor of the
+  // trigger element (in addition to setting it on the trigger element itself). Useful to keep an
+  // element highlighted while an associated menu is open.
+  parentSelectorToMark?: string;
```

Comment:
The `ctl.setOpenClass()` I added initially makes it possible to highlight things I wanted, but really awkward. E.g. sometimes the same menu is opened from two spots (e.g. a doc menu and pinned-doc menu), and it differs which element (if any) to highlight. Having it as an option to menu() (rather than something that must be done in the callback) makes it much more convenient.